index.js

```diff
@@ -414,5 +414,5 @@ function computeFnv1a32Hash (buf) {
     h ^= buf[i] & 0xff
     h += (h << 1) + (h << 4) + (h << 7) + (h << 8) + (h << 24)
   }
-  return h
+  return h & h
```

Comment:
Why this?

Reply:
Hash only matched for that one block tested, other ones I tested didnt, not sure why this works either, I changed the test to block that didnt match. Also checked hashing online "hello" and with h & h it matched

Reply:
If you're not sure then please revert. The implication here is that that has a substantive effect and if it does it's not obvious to me (bitwise AND on two numbers yields the number itself, it does not turn signed to unsigned or anything like that). The only effect I can think would be to turn a floating point into an integer, but getting a float is not possible in the operation here.

If it breaks without it then that needs some investigation because it doesn't seem to make sense.

Reply:
> If you're not sure then please revert. The implication here is that that has a substantive effect and if it does it's not obvious to me (bitwise AND on two numbers yields the number itself, it does not turn signed to unsigned or anything like that). The only effect I can think would be to turn a floating point into an integer, but getting a float is not possible in the operation here.
> 
> If it breaks without it then that needs some investigation because it doesn't seem to make sense.

I tested with my bot now and just returning "h" only works for some blocks, others hash collide
![image](https://github.com/PrismarineJS/prismarine-block/assets/19225467/9930a383-a4cb-4eef-92cc-5bd5aca1e602)

Reply:
Which ones?

Reply:
The one block I changed in tests fails without "h & h", could be used as an example (i think its soul_soil or other soul_...)

Reply:
Please add tests that are failing or show the collision. This duplication comment doesn't make any sense.

Reply:
I have added new tests that illustrate difference in outputted numbers

Reply:
Ah, this is because of dumb JavaScript behavior. The hash is exceeding the 32 bit boundary and because JS turns all floating point numbers into 32 bit integers when doing bitwise operations, `h & h` was effectively truncating the bits. This is very unclear so it should be moved to explicitly mask the number to 0xFF FF FF FF